app/channel_category.go

```diff
@@ -144,14 +144,135 @@ func (a *App) UpdateSidebarCategoryOrder(userId, teamId string, categoryOrder []
+			result := make(map[string]*model.SidebarCategoryWithChannels)
+			for _, category := range categories {
+				result[category.Id] = category
+			}
+
+			return result
+		}
+
+		updatedCategoriesById := makeCategoryMap(updatedCategories)
+		originalCategoriesById := makeCategoryMap(originalCategories)
+
+		for channelId, diff := range channelsDiff {
+			fromCategory := originalCategoriesById[diff.fromCategoryId]
+			toCategory := updatedCategoriesById[diff.toCategoryId]
+
+			if toCategory.Muted && !fromCategory.Muted {
+				channelsToMute = append(channelsToMute, channelId)
+			} else if !toCategory.Muted && fromCategory.Muted {
+				channelsToUnmute = append(channelsToUnmute, channelId)
+			}
+		}
+	}
+
+	for _, channelId := range channelsToMute {
+		_, err := a.setChannelMuted(channelId, userId, true)
+		if err != nil {
+			mlog.Error(
+				"Failed to mute channel to match category",
+				mlog.String("channel_id", channelId),
+				mlog.String("user_id", userId),
+				mlog.Err(err),
+			)
+		}
+	}
+
+	for _, channelId := range channelsToUnmute {
+		_, err := a.setChannelMuted(channelId, userId, false)
+		if err != nil {
+			mlog.Error(
+				"Failed to unmute channel to match category",
+				mlog.String("channel_id", channelId),
+				mlog.String("user_id", userId),
+				mlog.Err(err),
+			)
+		}
+	}
+}
+
+type categoryChannelDiff struct {
+	fromCategoryId string
+	toCategoryId   string
+}
+
+func diffChannelsBetweenCategories(updatedCategories []*model.SidebarCategoryWithChannels, originalCategories []*model.SidebarCategoryWithChannels) map[string]*categoryChannelDiff {
+	// mapChannelIdsToCategories returns a map of channel IDs to the IDs of the categories that they're a member of.
+	mapChannelIdsToCategories := func(categories []*model.SidebarCategoryWithChannels) map[string]string {
+		result := make(map[string]string)
+		for _, category := range categories {
+			for _, channelId := range category.Channels {
+				result[channelId] = category.Id
+			}
+		}
+
+		return result
+	}
+
+	updatedChannelIdsMap := mapChannelIdsToCategories(updatedCategories)
+	originalChannelIdsMap := mapChannelIdsToCategories(originalCategories)
+
+	// Check for any channels that have changed categories. Note that we don't worry about any channels that have moved
+	// outside of these categories since that heavily complicates things and doesn't currently happen in our apps.
+	channelsDiff := make(map[string]*categoryChannelDiff)
+	for channelId, originalCategoryId := range originalChannelIdsMap {
+		updatedCategoryId := updatedChannelIdsMap[channelId]
+
+		if originalCategoryId != updatedCategoryId && updatedCategoryId != "" {
+			channelsDiff[channelId] = &categoryChannelDiff{originalCategoryId, updatedCategoryId}
+		}
+	}
+
+	return channelsDiff
+}
+
+func getChannelIdsRemovedFromCategory(originalCategory *model.SidebarCategoryWithChannels, updatedCategory *model.SidebarCategoryWithChannels) []string {
```

Comment:
This function is not used anywhere. Any reason why this is kept?

Reply:
Oh yeah, I'll remove that. I meant to use it, but then it got replaced by `diffChannelsBetweenCategories`